File: app/src/main/java/com/boost/espressotest/presentation/screen/main/di/MainModule.java
package com.boost.espressotest.presentation.screen.main.di;

import com.boost.espressotest.domain.ProductRepository;
import com.boost.espressotest.presentation.screen.main.presenter.MainPresenter;

import dagger.Module;
import dagger.Provides;

/**
 * @author PerSpiKyliaTor on 19.01.18.
 */

@Module
public class MainModule {

    @Provides
    @MainScope
    MainPresenter provideMainPresenter(ProductRepository productRepository) {
        return new MainPresenter(productRepository);
    }
}

File: app/src/main/java/com/boost/espressotest/presentation/screen/main/di/MainModule.kt
package com.boost.espressotest.presentation.screen.main.di

import com.boost.espressotest.domain.ProductRepository
import com.boost.espressotest.presentation.screen.main.presenter.MainPresenter

import dagger.Module
import dagger.Provides

@Module
class MainModule {

    @Provides
    @MainScope
    internal fun provideMainPresenter(productRepository: ProductRepository): MainPresenter {
        return MainPresenter(productRepository)
    }
}
